src/logger.js

```diff
@@ -233,8 +252,15 @@ class Logger extends Writable {
       }
       cb();
     }).catch(err => {
-      this.emit(errorEvent, err);
-      this.debugLogger.log(`Error: ${err}`);
+      if (err instanceof ReconnectFailedError) {
+        this.ringBuffer.read(); // this connection failed, shift the buffer
```

Comment:
we are losing an event here. read() will return the ext event in the buffer by shifting it.